pkgs/servers/monitoring/karma/default.nix

```
@@ -1,37 +1,36 @@
-{ pkgs
-, lib
+{ lib
 , stdenv
 , buildGoModule
-, callPackage
 , fetchFromGitHub
+, fetchNpmDeps
 , nixosTests
-, nodejs_18
+, nodejs
+, npmHooks
 }:
 
-let
-  uiNodeDependencies = (import ./node-composition.nix {
-    inherit pkgs;
-    inherit (stdenv.hostPlatform) system;
-    # pin nodejs version
-    nodejs = nodejs_18;
-   }).nodeDependencies;
-in
-
 buildGoModule rec {
   pname = "karma";
-  version = "0.114";
+  version = "0.115";
 
   src = fetchFromGitHub {
     owner = "prymitive";
     repo = "karma";
     rev = "v${version}";
-    hash = "sha256-ZstBumK2ywwdr1ksMN7P8mHdYUiMOpfpYnvt0v0Io4w=";
+    hash = "sha256-SW/nmJcSk/LmoKLuD5stsSaRGaJctl6hVSODNCT9i64=";
   };
 
-  vendorHash = "sha256-ZsXPA4KyKbc/bwkidyHNDg62mE8KlE+yIssOBZLmHVg=";
+  vendorHash = "sha256-Y55AaB8KRV+Tq/Trg1BOOwziyt+yJ2b3iVYA6bDebQY=";
+
+  npmDeps = fetchNpmDeps {
+    src = "${src}/ui";
+    hash = "sha256-/L+eU0xwaopL2im9epiZiZ23dUqJ+3OwhWw/rIZC6hI=";
+  };
+
+  npmRoot = "ui";
 
   nativeBuildInputs = [
-    nodejs_18
+    nodejs
+    npmHooks.npmConfigHook
   ];
 
```

Thank you, this looks so, so much cleaner than before! For whatever reason, npmConfigHook was not detecting the npm dependencies on my system. It gave me the following error:

```
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/smx27hhdfy347sglcbm34rxh4fslsqa7-karma-0.115-go-modules.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/xfv9rds55ba8zx083sqpdb2h2qbl3j2d-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing npmConfigHook
Configuring npm

ERROR: no dependencies were specified
Hint: set `npmDeps` if using these hooks individually. If this is happening with `buildNpmPackage`, please open an issue.
```


Implementing an override from https://github.com/NixOS/nixpkgs/pull/229953 seemed to do the trick, at least on my system.

```suggestion

  overrideModAttrs = oldAttrs: {
    nativeBuildInputs = lib.filter (drv: drv != npmHooks.npmConfigHook) oldAttrs.nativeBuildInputs;
    preBuild = null;
  };

```

---

Oh that makes sense.